src/components/sign-attributes/sign-attribute-pane.vue

```diff
@@ -0,0 +1,259 @@
+            op.redo();
+            ops.push(op);
+        }
+        this.$state.eventBus.emit('new-bulk-operations', ops);
+        this.keepOpen = false;
+        (this.$refs.attributesMenu as BDropdown).hide();
+    }
+
+    private onAddAttributesMenuOpen() {
+        this.attributesMenu = this.prepareAttributesMenu();
+    }
+
+    private prepareAttributesMenu(): AttributeDTO[] {
+        if (!this.selectedSignsInterpretation.length) {
+            return [];
+        }
+
+        const filteredAttributes: AttributeDTO[] = [];
+        const attributesSet: Set<number> = new Set<number>();
+        const attributesValuesSet: Set<number> = new Set<number>();
+
+        for (const si of this.selectedSignsInterpretation) {
+            for (const attribute of si.attributes) {
+                attributesSet.add(attribute.attributeId);
+                attributesValuesSet.add(attribute.attributeValueId);
+            }
+        }
+
+        for (const attributeMeta of this.attributesMetadata) {
+
+            const attributeCopy = {...attributeMeta};
+            // check repeatable
+            if (attributesSet.has(attributeMeta.attributeId) && !attributeMeta.repeatable) {
+                continue;
+            }
+
+            // multiple: only batchEditable
+            if (this.selectedSignsInterpretation.length > 1 && !attributeMeta.batchEditable) {
+                continue;
+            }
+
+            // single: only editable
+            if (this.selectedSignsInterpretation.length === 1 && !attributeMeta.editable) {
+                continue;
+            }
+
+            // repeatable: remove existing values
+            if (attributesSet.has(attributeMeta.attributeId) && attributeMeta.repeatable) {
+                for (const attributeValue of attributeMeta.values) {
+                    if (attributesValuesSet.has(attributeValue.id)) {
+                        const idx = attributeCopy.values.findIndex(value => value.id === attributeValue.id);
+                        attributeCopy.values.splice(idx, 1);
+                    }
+                }
+            }
+
+            filteredAttributes.push(attributeCopy);
+        }
+
+        return filteredAttributes;
+    }
+
+    private onValuesMenuShow() {
+        this.keepOpen = true;
+    }
+
+    private onValuesMenuHide() {
+        this.keepOpen = false;
+        // (this.$refs.attributesMenu as BDropdown).hide();
+    }
+
+    private onAttributesMenuHide(event: BvEvent) {
+        if (this.keepOpen) {
+            event.preventDefault();
+        }
+    }
+}
+</script>
+
+<style lang="scss" scoped>
+.attributes {
+    height: 10%;
+    border-top: 3px solid #6c757d;
+    padding: 10px;
+    ul {
+        list-style: none;
+    }
+}
+.btn-group > .btn,
+.btn-group-vertical > .btn {
+    position: relative;
+    -ms-flex: 1 1 auto;
+    flex: 1 1 auto;
+    padding: 3px;
+    background-color: white;
+    color: black;
+    border: 0px;
+}
+</style>
+<style lang="scss">
```

Comment:
Is there a good reason for a scoped css block and then another non-scoped css block?  I though we said we would try to just used scoped ones.

Reply:
No, I think the two blocks were added by mistake. I've combined them into one.